analysis/deciles_charts.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+import pathlib
+import re
+
+import pandas
+from ebmdatalab import charts
+
+
+MEASURE_FNAME_REGEX = re.compile(r"measure_(?P<id>\w+)\.csv")
+
+
+def _get_denominator(measure_table):
+    return measure_table.columns[-3]
+
+
+def _get_group_by(measure_table):
+    return list(measure_table.columns[:-4])
+
+
+def get_measure_tables(path):
+    if not path.is_dir():
+        raise AttributeError()
+
+    for sub_path in path.iterdir():
+        if not sub_path.is_file():
+            continue
+
+        measure_fname_match = re.match(MEASURE_FNAME_REGEX, sub_path.name)
+        if measure_fname_match is not None:
+            # The `date` column is assigned by the measures framework.
+            measure_table = pandas.read_csv(sub_path, parse_dates=["date"])
+
+            # We can reconstruct the parameters passed to `Measure` without
+            # the study definition.
+            measure_table.attrs["id"] = measure_fname_match.group("id")
+            measure_table.attrs["denominator"] = _get_denominator(measure_table)
+            measure_table.attrs["group_by"] = _get_group_by(measure_table)
+
+            yield measure_table
+
+
+def drop_zero_denominator_rows(measure_table):
+    mask = measure_table[measure_table.attrs["denominator"]] > 0
+    return measure_table[mask].reset_index(drop=True)
+
+
+def get_deciles_chart(measure_table):
+    return charts.deciles_chart(measure_table, period_column="date", column="value")
```

Comment:
If we have changes to the ebmdatalab decile charts code, do you think we will push updates to the library? Or have our own code here? In relation to [https://github.com/opensafely-core/cohort-extractor/issues/759](url), we might want to be about to output the intermediate deciles tables for output checking.

Reply:
I think that we'll probably end up replacing that implementation with a new implementation, in this module. Writing out intermediate deciles tables would be one reason for replacing implementations (in this case, we could revert 31832fa0a2e8caace78ac78d34bf1187f98b1ea6). I decided to fall back to `charts.deciles_chart` because it's the canonical implementation.